### PR TITLE
[MODGOBI-166]-added translater

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose

https://issues.folio.org/browse/MODGOBI-166
## Approach

Move acq unit mapping logic in model to the translation object
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json [hunk 1]
```diff
@@ -1,313 +1,316 @@
...
+    {
+      "field": "ACQUISITION_UNIT",
+      "dataSource": {
+        "default": "gobi_acq_unit_name",
+        "translation": "lookupAcquisitionUnitIdsByName",
+        "translateDefault": true
+      }
+    },
+    {
+      "field": "VENDOR_ACCOUNT",
```
Comment (SerhiiNosko) [inline src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json:20]: There sould be only 1 field ACQUISITION_UNIT that mapped to one of 2 available translators. 
VENDOR_ACCOUNT field should not be me mapped with lookupAcquisitionUnitIdsByAccount translator.

File: src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json [hunk 2]
```diff
@@ -1,313 +1,316 @@
...
+  "mappings": [
+    {
+      "field": "ACCESS_PROVIDER",
+      "dataSource": {
+        "from": "//PurchaseOption/VendorPOCode",
+        "translation": "lookupOrganization"
+      }
+    },
+    {
+      "field": "ACQUISITION_UNIT",
```
Comment (SerhiiNosko) [inline src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json:12]: We already have karate test for gobi, please run it to make sure that everething is okay, also add additaional assert if needed to that karate test that checks that acq units correctly assigned.
If you have issue with local run - let me know, we will try to fix

File: src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json (hunk 1 above)
Comment (SinghAdes) [reply src/test/resources/MappingHelper/Custom_ListedElectronicSerial.json]: fixed.

File: src/main/java/org/folio/gobi/LookupService.java [hunk 3]
```diff
@@ -238,6 +240,15 @@ public CompletableFuture<String> lookupAcquisitionUnitDefault(String data) {
 
   }
 
+  public CompletableFuture<Object> lookupAcquisitionUnitIdsByAccount(String data) {
+
+    return lookupOrganization(ORGANIZATION_NAME).thenApply(org -> org.getAccounts().stream()
+      .filter(acc -> HelperUtils.normalizeSubAccout(acc.getAccountNo()).equals(HelperUtils.normalizeSubAccout(data)))
```
Comment (okolawole-ebsco) [inline src/main/java/org/folio/gobi/LookupService.java:246]: nitpick: `HelperUtils.normalizeSubAccount(data)` should be the same value for execution of this lambda. Consider saving its output into a variable and using that instead.

**Comment by okolawole-ebsco** [comment]
Any tests possible for this?

File: src/main/java/org/folio/gobi/LookupService.java (hunk 3 above)
Comment (grabsefx) [reply src/main/java/org/folio/gobi/LookupService.java:246]: actually no need to normalize values of stored organization. Only gobi request can contain subacount with "-" delimeter

File: src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java [hunk 4]
```diff
@@ -985,6 +984,7 @@ private void verifyRequiredFieldsAreMapped(CompositePurchaseOrder compPO){
     assertNotNull(poLine.getOrderFormat());
     assertNotNull(poLine.getSource());
     assertNotNull(poLine.getTitleOrPackage());
+    assertNotNull(compPO.getAcqUnitIds());
```
Comment (grabsefx) [inline src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java:987]: please change this assertion to check if this array is not empty

Comment (SinghAdes) [reply src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java]: done.

File: src/main/java/org/folio/gobi/LookupService.java (hunk 3 above)
Comment (SinghAdes) [reply src/main/java/org/folio/gobi/LookupService.java]: fixed.

**Comment by SinghAdes** [comment]
> Any tests possible for this?

we have a test for this :-https://github.com/folio-org/mod-gobi/blob/master/src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java#:~:text=void%20testPostGobiOrdersCustomPOListedElectronicSerial()%20throws%20Exception

File: src/main/java/org/folio/gobi/LookupService.java (hunk 3 above)
Comment (grabsefx) [reply src/main/java/org/folio/gobi/LookupService.java]: However `data` variable needs to be normalized

Comment (SinghAdes) [reply src/main/java/org/folio/gobi/LookupService.java]: Ah okay , get it. 
fixed

**Review by grabsefx — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**